pkg/csi/service/osutils/windows_os_utils.go

```diff
@@ -364,19 +364,30 @@ func (osUtils *OsUtils) IsTargetInMounts(ctx context.Context, path string) (bool
 // Defaults to nfs4 for file volume and ntfs for block volume when empty string
 // is observed. This function also ignores default ext4 fstype supplied by
 // external-provisioner when none is specified in the StorageClass
-func (osUtils *OsUtils) GetVolumeCapabilityFsType(ctx context.Context, capability *csi.VolumeCapability) string {
+func (osUtils *OsUtils) GetVolumeCapabilityFsType(ctx context.Context,
+	capability *csi.VolumeCapability) (string, error) {
 	log := logger.GetLogger(ctx)
 	fsType := strings.ToLower(capability.GetMount().GetFsType())
-	log.Debugf("FsType received from Volume Capability: %q", fsType)
+	log.Infof("FsType received from Volume Capability: %q", fsType)
 	isFileVolume := common.IsFileVolumeRequest(ctx, []*csi.VolumeCapability{capability})
-	if isFileVolume && (fsType == "" || fsType == "ext4") {
-		log.Infof("empty string or ext4 fstype observed for file volume. Defaulting to: %s", common.NfsV4FsType)
-		fsType = common.NfsV4FsType
-	} else if !isFileVolume && fsType == "" {
-		log.Infof("empty string fstype observed for block volume. Defaulting to: %s", common.NTFSFsType)
-		fsType = common.NTFSFsType
-	}
-	return fsType
+	if isFileVolume {
+		// Volumes with RWM or ROM access modes are not supported on Windows
+		return "", logger.LogNewErrorCode(log, codes.FailedPrecondition,
```

Comment:
if the user has not specified NTFS file system in the storage class, and the external provisioner has supplied default ext4, we are not defaulting to NTFS and failing with the error "vSAN file service volume can not be mounted on windows node".
Can we test this on windows setup? We do not want to mandate users to re-create storage class and explicitly mention ntfs fstype in the storageclass for windows workload.

Reply:
I see the code below this change. Ignore this comment.